library/pod-security-policy/host-filesystem/template.yaml

```diff
@@ -39,110 +39,145 @@ spec:
-            not allowedHostPath.readOnly == true
-        }
-
-        input_hostpath_allowed(allowedPaths, volume) {
-            allowedHostPath := allowedPaths[_]
-            path_matches(allowedHostPath.pathPrefix, volume.hostPath.path)
-            allowedHostPath.readOnly
-            not writeable_input_volume_mounts(volume.name)
-        }
-
-        writeable_input_volume_mounts(volume_name) {
-            container := input_containers[_]
-            mount := container.volumeMounts[_]
-            mount.name == volume_name
-            not mount.readOnly
-        }
-
-        # This allows "/foo", "/foo/", "/foo/bar" etc., but
-        # disallows "/fool", "/etc/foo" etc.
-        path_matches(prefix, path) {
-            a := path_array(prefix)
-            b := path_array(path)
-            prefix_matches(a, b)
-        }
-        path_array(p) = out {
-            p != "/"
-            out := split(trim(p, "/"), "/")
-        }
-        # This handles the special case for "/", since
-        # split(trim("/", "/"), "/") == [""]
-        path_array("/") = []
-
-        prefix_matches(a, b) {
-            count(a) <= count(b)
-            not any_not_equal_upto(a, b, count(a))
-        }
-
-        any_not_equal_upto(a, b, n) {
-            a[i] != b[i]
-            i < n
-        }
-
-        input_hostpath_volumes[v] {
-            v := input.review.object.spec.volumes[_]
-            has_field(v, "hostPath")
-        }
-
-        # has_field returns whether an object has a field
-        has_field(object, field) = true {
-            object[field]
-        }
-        input_containers[c] {
-            c := input.review.object.spec.containers[_]
-        }
-
-        input_containers[c] {
-            c := input.review.object.spec.initContainers[_]
-        }
-
-        input_containers[c] {
-            c := input.review.object.spec.ephemeralContainers[_]
-        }
-      libs:
-        - |
-          package lib.exclude_update
-
-          is_update(review) {
-              review.operation == "UPDATE"
-          }
+      code:
+      - engine: K8sNativeValidation
+        source: 
+          variables:
+          - name: containers
+            expression: 'has(variables.anyObject.spec.containers) ? variables.anyObject.spec.containers : []'
+          - name: initContainers
+            expression: 'has(variables.anyObject.spec.initContainers) ? variables.anyObject.spec.initContainers : []'
+          - name: ephemeralContainers
+            expression: 'has(variables.anyObject.spec.ephemeralContainers) ? variables.anyObject.spec.ephemeralContainers : []'
+          - name: allContainers
+            expression: 'variables.containers + variables.initContainers + variables.ephemeralContainers'
+          - name: allowedHostPaths
+            expression: |
+              has(variables.params) ? (has(variables.params.allowedHostPaths) ? variables.params.allowedHostPaths : []) : []
```

Comment:
nit: If you use !has() the chaining should look a bit cleaner and will end with the user-input variable value